meson.build

```diff
@@ -0,0 +1,82 @@
+project('re2', 'cpp',
+        version : '20201101',
+        license : 'BSD-3-Clause',
+        default_options : [
+          'cpp_std=c++11',
+        ])
+
+threads_dep = dependency('threads')
+
+cpp = meson.get_compiler('cpp')
+
+if cpp.get_id() == 'msvc'
+  cpp_args = [
+    '/wd4100',
+    '/wd4201',
+    '/wd4456',
+    '/wd4457',
+    '/wd4702',
+    '/wd4815',
+    '/utf-8',
+  ]
+  cpp_args = cpp.get_supported_arguments(args)
+  add_project_arguments(cpp_args, language: 'cpp')
+endif
+
+if target_machine.system() == 'windows'
+  cpp_args = [
+    '-DUNICODE',
+    '-D_UNICODE',
+    '-DSTRICT',
+    '-DNOMINMAX',
+    '-D_CRT_SECURE_NO_WARNINGS',
+    '-D_SCL_SECURE_NO_WARNINGS',
+  ]
+  add_project_arguments(cpp_args, language: 'cpp')
+endif
+
+re2_sources = files(
+  're2/bitstate.cc',
+  're2/compile.cc',
+  're2/dfa.cc',
+  're2/filtered_re2.cc',
+  're2/mimics_pcre.cc',
+  're2/nfa.cc',
+  're2/onepass.cc',
+  're2/parse.cc',
+  're2/perl_groups.cc',
+  're2/prefilter.cc',
+  're2/prefilter_tree.cc',
+  're2/prog.cc',
+  're2/re2.cc',
+  're2/regexp.cc',
+  're2/set.cc',
+  're2/simplify.cc',
+  're2/stringpiece.cc',
+  're2/tostring.cc',
+  're2/unicode_casefold.cc',
+  're2/unicode_groups.cc',
+  'util/rune.cc',
+  'util/strutil.cc',
+)
+
+includes = include_directories('.')
+
+libre2 = library(
```

Comment:
This should declare soname 9, for the event it is built with `-Ddefault_library=shared`.

Reply:
Where did you get the soname from? I see it in the Makefile but not in the CMake definition.

Reply:
https://github.com/google/re2/issues/274#issuecomment-683440146

> For the record, RE2 supports CMake mostly for the sake of users who don't have GNU make (e.g. Visual Studio users, Xcode users) and only partly – and unenthusiastically – for the sake of CMake users. Having said that, I actually want to help here. :)

https://github.com/google/re2/issues/259#issuecomment-629737504
> Ahh. So you are a Linux distribution package maintainer. That information is helpful to have – and having it earlier would have prevented me from suggesting CMake because that isn't intended for Linux distributions.

Indeed, distros exclusively build using the Makefile, while the CMake definition is a second-class citizen used by niche users.

Reply:
Thank you. Looking at the Makefile now